From 85a1e8b7a4f919a49c9766fce76b1bde5e3e0c64 Mon Sep 17 00:00:00 2001
From: Haiqing Bai <Haiqing.Bai@windriver.com>
Date: Fri, 30 Aug 2019 06:01:27 +0000
Subject: [PATCH] net: core: Fixed warning:
 'proc_dointvec_minmax_bpf_restricted' defined but not used

For 'proc_dointvec_minmax_bpf_restricted' is static function and only
is registered in 'static struct ctl_table net_core_table[]' if both
CONFIG_BPF_JIT and CONFIG_HAVE_EBPF_JIT are enabled, so adding ifdef to
function definition of 'proc_dointvec_minmax_bpf_restricted' to avoid:
"net/core/sysctl_net_core.c:285:1: warning: 'proc_dointvec_minmax_bpf_restricted'
defined but not used [-Wunused-function]"

Signed-off-by: Haiqing Bai <Haiqing.Bai@windriver.com>
---
 net/core/sysctl_net_core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/core/sysctl_net_core.c b/net/core/sysctl_net_core.c
index d67ec17f2cc8..eb5156619437 100644
--- a/net/core/sysctl_net_core.c
+++ b/net/core/sysctl_net_core.c
@@ -281,6 +281,7 @@ static int proc_dointvec_minmax_bpf_enable(struct ctl_table *table, int write,
 	return ret;
 }
 
+#ifdef CONFIG_HAVE_EBPF_JIT
 static int
 proc_dointvec_minmax_bpf_restricted(struct ctl_table *table, int write,
 				    void __user *buffer, size_t *lenp,
@@ -291,6 +292,7 @@ proc_dointvec_minmax_bpf_restricted(struct ctl_table *table, int write,
 
 	return proc_dointvec_minmax(table, write, buffer, lenp, ppos);
 }
+#endif
 
 static int
 proc_dolongvec_minmax_bpf_restricted(struct ctl_table *table, int write,
-- 
2.17.1

